AppKit/CPWindow/CPWindow.j

```diff
@@ -1958,6 +1967,79 @@ CPTexturedBackgroundWindowMask
+                        var aTrackingArea   = [trackingAreas objectAtIndex:j],
+                        trackingOptions = [aTrackingArea options];
+                        
+                        if (trackingOptions & CPTrackingEnabledDuringMouseDrag)
+                            
+                            if ((trackingOptions & CPTrackingActiveAlways) ||
+                                ((trackingOptions & CPTrackingActiveInKeyWindow) && ([self isKeyWindow])) ||
+                                ((trackingOptions & CPTrackingActiveWhenFirstResponder) && ([self isKeyWindow]) && (_firstResponder == aView)))
+                                
+                                if (CGRectContainsPoint(((trackingOptions & CPTrackingInVisibleRect) ? [aView visibleRect] : [aTrackingArea rect]), convertedPoint))
+                                {
+                                    [mouseEnteredStack addObject:aTrackingArea];
+                                    
+                                    if (![_mouseEnteredStack containsObject:aTrackingArea])
+                                        
+                                        // Mouse was not in this rect so it's a mouseEntered
+                                        
+                                        if (trackingOptions & CPTrackingMouseEnteredAndExited)
+                                            
+                                            [[aTrackingArea owner] mouseEntered:[CPEvent enterExitEventWithType:CPMouseEntered
+                                                                                                       location:point
+                                                                                                  modifierFlags:[anEvent modifierFlags]
+                                                                                                      timestamp:[anEvent timestamp]
+                                                                                                   windowNumber:_windowNumber
+                                                                                                        context:nil
+                                                                                                    eventNumber:-1
+                                                                                                   trackingArea:aTrackingArea]];
+                                }
+                }
+            }
+            
+            // Search now for exited views (were in _mouseEnteredStack but no more in mouseEnteredStack)
+            
+            for (var i = 0, nb = [_mouseEnteredStack count]; i < nb; i++)
+            {
+                var aTrackingArea   = [_mouseEnteredStack objectAtIndex:i],
+                    trackingOptions = [aTrackingArea options];
+
+                if (trackingOptions & CPTrackingEnabledDuringMouseDrag)
+
+                    if (![mouseEnteredStack containsObject:aTrackingArea])
```

Comment:
I've checked for using dictionary instead of arrays but I don't think we would gain a lot as : 
- we deal here with only a few values
- we could avoid loops in loops (by avoiding containsObject) but when parsing for all values of a stack, we would have to use a dictionary enumerator, which represents a massive overhead against simple array parsing.

Indeed I've already optimized (in my main app) large arrays (more than 30.000 items) by using dictionaries (way more efficient in some operations) but I had to put hands in the engine (BTW, I don't know if this is a valid expression in English) and directly deal with buckets. But this is not useful here (I think so but could be wrong).